### PR TITLE
Victory screen

### LINKED ISSUE #35: Base screen for Victory + Highscore
Game can end

### PR DESCRIPTION
**Pull request by ivantjh**
- Create Classic & Time Attack game modes.
- Implement Victory and High Score screen
- Read and save highscore from json

Scene layers can now communicate through `sceneData` struct. After new highscore is added, all layers will be popped to reveal the base (menu) layer.

Fixes #35

### CODE REVIEW

File: src/classic.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	
+	if (leftPaddleScore == pongstarNS::POINTS_TO_WIN || rightPaddleScore == pongstarNS::POINTS_TO_WIN)
+		winnerFound = true;
+	
+	if (winnerFound) {
+		nextSceneType = sceneNS::VICTORY;
+		sceneData.modePlayed = sceneNS::GM_CLASSIC;
+		sceneData.winner = leftPaddleScore == pongstarNS::POINTS_TO_WIN ? sceneNS::W_LEFT : sceneNS::W_RIGHT;
+		sceneData.newHighScore.score = elapsedTime;
+	}
```
Comment (chesterhow) [inline src/classic.cpp:21]: Why not just do this is the first `if` claus?

File: src/pongstar.cpp [hunk 2]
```diff
@@ -57,15 +57,25 @@ void Pongstar::initialize(HWND hwnd) {
...
-	Menu* menu = new Menu(input, fontManager);
-	menu->initialize();
-	gameStack->push(menu);
+	sceneNS::SceneData sd = sceneNS::SceneData();
+	//sd.modePlayed = sceneNS::GM_CLASSIC;
+	//sd.winner = paddleNS::RIGHT;
+	//sd.newHighScore.name = "LALAA";
+	//sd.newHighScore.score = 21000;
+	//sd.hsDisplayMode = sceneNS::HS_BOTH;
+	//sd.winner = sceneNS::W_DRAW;
```
Comment (chesterhow) [inline src/pongstar.cpp]: Whats all these

File: src/scene.h [hunk 3]
```diff
@@ -5,20 +5,54 @@
...
+	enum GAME_MODE {
+		GM_CLASSIC, GM_TIME_ATK
+	};
+
+	enum WINNER {
+		W_LEFT, W_RIGHT, W_DRAW
+	};
+
+	struct NewHighScore {
+		int score;
```
Comment (chesterhow) [inline src/scene.h:24]: Why `int`? what about classic mode's timing highscores

File: src/textureManager.h [hunk 4]
```diff
@@ -32,7 +31,7 @@ class TextureManager {
 	UINT getHeight() const { return height; }
 
 	// Initialize the textureManager
-	// Pre: *g points to Graphics object
+	// Pre: *g points to Graphicso bject
```
Comment (chesterhow) [inline src/textureManager.h]: mhhmm

**Review by chesterhow — CHANGES_REQUESTED** [review]

File: src/pongstar.cpp (hunk 2 above)
Comment (ivantjh) [reply src/pongstar.cpp]: For testing, removing them

File: src/scene.h (hunk 3 above)
Comment (ivantjh) [reply src/scene.h:24]: Classic mode is stored as milliseconds, which is also int.

File: src/textureManager.h (hunk 4 above)
Comment (ivantjh) [reply src/textureManager.h]: Changed

**Review by chesterhow — APPROVED** [review]

**Pull request merged**